inference-engine/src/legacy_api/src/network_serializer_v7.cpp

```diff
@@ -254,90 +255,9 @@ void UpdateStdLayerParams(const CNNLayer::Ptr& layer) {
-        for (const auto & output : node->outData) {
-            for (const auto &consumer : getInputTo(output)) {
-                consumers.push_back(consumer.second);
-            }
-        }
-        return consumers;
-    };
-    auto bfs = [&used, &ordered, &get_consumers](const CNNLayerPtr& start_node, bool traverse_via_outputs = false) {
-        if (!start_node) return;
-        std::deque<CNNLayerPtr> q;
-        q.push_front(start_node);
-        while (!q.empty()) {
-            auto node = q.front();
-            q.pop_front();
-            if (used.insert(node->name).second) {
-                ordered.push_back(node);
-            }
-
-            // Traverse via inputs
-            for (const auto & input : node->insData) {
-                auto locked_input = input.lock();
-                if (!locked_input) {
-                    THROW_IE_EXCEPTION << "insData for " << node->name << " is not valid.";
-                }
-                if (auto next_node = getCreatorLayer(locked_input).lock()) {
-                    if (!used.count(next_node->name)) {
-                        // Check that all consumers were used
-                        bool all_consumers_used(true);
-                        for (const auto & consumer : get_consumers(next_node)) {
-                            if (!used.count(consumer->name)) all_consumers_used = false;
-                        }
-                        if (all_consumers_used) {
-                            q.push_front(next_node);
-                        }
-                    }
-                }
-            }
-
-            // Traverse via outputs
-            if (traverse_via_outputs) {
-                for (const auto &consumer : get_consumers(node)) {
-                    if (!used.count(consumer->name)) {
-                        q.push_front(consumer);
-                    }
-                }
-            }
-        }
-    };
-
-    // First we run bfs starting from outputs that provides deterministic graph traverse
-    for (const auto & output : outputs) {
-        if (!used.count(output.first)) {
-            bfs(getCreatorLayer(output.second).lock());
-        }
-    }
-
-    // For cases when graph has no outputs we start bfs from inputs to ensure topological sort
-    for (const auto & input : inputs) {
-        const auto data_ptr = input.second->getInputData();
-        for (const auto & consumer : getInputTo(data_ptr))
-        if (!used.count(consumer.first)) {
-            bfs(consumer.second, true);
-        }
-    }
-
-    std::reverse(ordered.begin(), ordered.end());
-    return ordered;
-}
-
 std::size_t FillXmlDoc(const InferenceEngine::ICNNNetwork& network, pugi::xml_document& doc,
                        const bool execGraphInfoSerialization, const bool dumpWeights) {
-    const std::vector<CNNLayerPtr> ordered = TopologicalSort(network);
+    const std::vector<CNNLayerPtr> ordered = InferenceEngine::details::CNNNetSortTopologically(network);
```

Comment:
why have you changed it? 

@GlebKazantaev as far as I remember, you have created `TopologicalSort` for some reasons while `InferenceEngine::details::CNNNetSortTopologically` does not work here..

Reply:
1) Because there are 2 implementations of the same function.
2) the implementation in the serializer is not correct – I have a network which is not fully serialized with implementation.

Reply:
Do you recommend to reverse change in serializer?

Reply:
I want @GlebKazantaev to comment

Reply:
I tried to remember the reason why this graph traversal was implemented and why `CNNNetSortTopologically` wasn't used but I can't. As it affects only legacy serialization I suggest to keep only `CNNNetSortTopologically` until we get some troubles.  
And BTW @ilya-lavrenov  I got some build troubles when I turned on V7 serialization flag.